code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -62,8 +60,9 @@
 		return baby
 
 /mob/living/simple_animal/pet/cat/Runtime/death()
-	if(can_die() && !memory_saved)
-		Write_Memory(1)
+	if(can_die())
```

Comment:
Same here use the `. = ..()` as stuff to check if they actually died.

Reply:
Same here